File: 1122Programming/leetcode/1. Two Sum/ans.cs
/*
 *
 * https://leetcode.com/problems/two-sum/
 * Problem ID: 1. Two Sum
 * 
 * Author: asd7766zxc
 * 
 */


public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var mapping = nums.Select((item, index) => new {item,index}).ToLookup(pair => pair.item, pair => pair.index);
        return nums.Select((item, index) => new { item, index }).Where((first,ind)=> mapping.Contains(target - first.item) && mapping[target-first.item].First()!=ind).Select((first, ind) => new int[]{first.index,mapping[target - first.item].First() }).FirstOrDefault();
    }
}

File: 1122Programming/leetcode/1. Two Sum/ans.kt
/*
 *
 * https://leetcode.com/problems/two-sum/
 * Problem ID: 1. Two Sum
 * 
 * Author: revival0728
 * 
 */

class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val arr = Array(nums.size) { Pair(nums[it], it) }
        arr.sortBy { it.first }
        return arr.run {
            val ans = Pair(0, this.lastIndex).toList().toMutableList()
            var sum = this[ans[0]].first + this[ans[1]].first
            while(sum != target) {
                if(sum < target)
                    ++ans[0]
                else
                    --ans[1]
                sum = this[ans[0]].first + this[ans[1]].first
            }

            IntArray(2) { this[ans[it]].second }
        }
    }
}
